contracts/contracts/buyback/BaseBuyback.sol

```diff
@@ -135,110 +153,135 @@ abstract contract BaseBuyback is Initializable, Strategizable {
-            rewardsSource,
-            amountInForOGV,
-            minOGV,
-            _getSwapPath(ogv),
-            false
-        );
+        emit OTokenBuyback(oToken, tokenOut, minAmountOut, amountOut);
+    }
 
-        inputs[1] = abi.encode(
-            // Buyback contract receives the CVX to lock it on
-            // behalf of Strategist after the swap
-            address(this),
-            amountInForCVX,
-            minCVX,
-            _getSwapPath(cvx),
-            false
-        );
+    /**
+     * @dev Swaps `oTokenAmount` to OGV
+     * @param oTokenAmount Amount of OUSD/OETH to swap
+     * @param minOGV Minimum OGV to receive for oTokenAmount
+     * @param swapData 1inch Swap Data
+     */
+    function swapForOGV(
+        uint256 oTokenAmount,
+        uint256 minOGV,
+        bytes calldata swapData
+    ) external onlyGovernorOrStrategist nonReentrant {
+        (uint256 _ogvAmount, ) = _updateBuybackSplits();
+        require(_ogvAmount >= oTokenAmount, "Balance underflow");
+        require(rewardsSource != address(0), "RewardsSource contract not set");
+
+        // Subtract the amount to swap from net balance
+        ogvShare = _ogvAmount - oTokenAmount;
+
+        uint256 ogvReceived = _swapToken(ogv, oTokenAmount, minOGV, swapData);
 
-        // Transfer OToken to UniversalRouter for swapping
+        // Transfer OGV received to RewardsSource contract
         // slither-disable-next-line unchecked-transfer unused-return
-        IERC20(oToken).transfer(universalRouter, oTokenAmount);
+        IERC20(ogv).transfer(rewardsSource, ogvReceived);
+    }
 
-        // Execute the swap
-        IUniswapUniversalRouter(universalRouter).execute(
-            swapCommand,
-            inputs,
-            block.timestamp
-        );
+    /**
+     * @dev Swaps `oTokenAmount` to CVX
+     * @param oTokenAmount Amount of OUSD/OETH to swap
+     * @param minCVX Minimum CVX to receive for oTokenAmount
+     * @param swapData 1inch Swap Data
+     */
+    function swapForCVX(
+        uint256 oTokenAmount,
+        uint256 minCVX,
+        bytes calldata swapData
+    ) external onlyGovernorOrStrategist nonReentrant {
+        (, uint256 _cvxAmount) = _updateBuybackSplits();
+        require(_cvxAmount >= oTokenAmount, "Balance underflow");
 
-        // Uniswap's Universal Router doesn't return the `amountOut` values/
-        // So, the events just calculate the tokens received by doing a balance diff
-        emit OTokenBuyback(
-            oToken,
-            ogv,
-            amountInForOGV,
-            IERC20(ogv).balanceOf(rewardsSource) - ogvBalanceBefore
-        );
-        emit OTokenBuyback(
-            oToken,
-            cvx,
-            amountInForCVX,
-            IERC20(cvx).balanceOf(address(this)) - cvxBalanceBefore
-        );
+        // Subtract the amount to swap from net balance
+        cvxShare = _cvxAmount - oTokenAmount;
```

Comment:
This could use unsafe math since a check is made on line 252. You could also remove the require check,

Reply:
Done as well